### PR TITLE
SOLR-15204: Document bin solr zk and zkcli

### PR DESCRIPTION
**Pull request by epugh**
# Description

Prefer the bin/solr zk command where possible over zkcli.sh.

# Solution

Updating docs.  Maybe will also extend in seperate tickets adding more commands to bin/solr zk if needed???

# Tests

doc change only, manual testing!

# Checklist

Please review the following and check all that apply:

- [X ] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ X] I have created a Jira issue and added the issue ID to my pull request title.
- [ X] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [X ] I have developed this patch against the `master` branch.
- [ X] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ X] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

File: solr/solr-ref-guide/src/authentication-and-authorization-plugins.adoc [hunk 1]
```diff
@@ -70,17 +70,23 @@ Here is a more detailed `security.json` example. In this, the Basic authenticati
...
 ----
 
 Note that this example defines the `KerberosPlugin` for authentication. You will want to modify this section as appropriate for the plugin you are using.
 
-This example also defines `security.json` on the command line, but you can also define a file locally and upload it to ZooKeeper.
+Then use the `bin/solr zk` command to upload the file:
+
+[source,bash]
+----
+>bin/solr zk cp /security.json zk:security.json -z localhost:2181
```
Comment (janhoy) [inline solr/solr-ref-guide/src/authentication-and-authorization-plugins.adoc:88]: Don’t assume the file is at root `/`. Better with `./`?

**Review by janhoy — APPROVED** [review]
Good, just one comment

**Comment by epugh** [comment]
@ctargett I have added a table trying to list the two CLI tools, and what is there equivalent commands...   Do you think it helps or hurts the page?   Should I list the other ZK related commands that either tool has in the table, showing that some things you need tool A, and others you need Tool B?

I'm imanging that way in the future there is a single "guide" type page that is "Here is all the ways of interacting with Solr ecosystem, and here is the tool you use", versus todays "tool centric" page...

**Comment by epugh** [comment]
There is one page about enabling SSL that appears to only work with zkcli: https://lucene.apache.org/solr/guide/8_8/enabling-ssl.html#configure-zookeeper.   

I wanted to use the API to set the CLUSTERPROP, but then realized that this is being done with a non running solr, so no api.      Awkwardly, we then document the api call just below.   I think if you already have solr running.

Does this suggest that maybe having a `bin/solr zk clusterprop` would be good?

**Comment by epugh** [comment]
Other than that one question about maybe we should add a `bin/solr zk clusterprop`, I think this is done in terms of updating the ref guide, unless others see some examples?

**Review by cpoerschke — APPROVED** [review]

**Comment by ctargett** [comment]
The changes to `command-line-utilities.adoc` aren't going to work when my upcoming re-org branch is merged because it's being renamed to `zookeeper-utilities.adoc` - it was never about general command line utilities, only ever about using Solr's version of zkcli.sh so has always been misnamed. In that vein, I'd prefer not to start to repurpose it to become some general page about "command line utilities" - there is a page for `bin/solr` and we should prefer that, not muddy things further with another page we turn into a comparison of tools.

An ideal state IMO would be that we don't need to document Solr's version of the zkcli.sh script at all, and could entirely get rid of the ZK-centric page and instead focus attention and energy on `bin/solr` extensions/documentation.

**Comment by janhoy** [comment]
> Does this suggest that maybe having a `bin/solr zk clusterprop` would be good?

Yes, that is the main missing command I believe. Feel free to hack away at https://issues.apache.org/jira/browse/SOLR-10189

**Comment by epugh** [comment]
I agree about `command-line-utilities.adoc` having the wrong name, so excited about that new name.   I also agree that we'd like to eventually get rid of the ZK specific cli as well!    Heck, I'd rather take the good parts of our `zkcli.sh` and migrate them to the main Zookeeper projects `zkcli.sh` if that made sense!

So @ctargett if I roll back the changes to `command-line-utilities.adoc`, then does the other edits make sense to you?

**Comment by ctargett** [comment]
> if I roll back the changes to command-line-utilities.adoc, then does the other edits make sense to you?

Yes, +1 to those.

I agree ultimately on zkcli.sh and would also like to actually get rid of the page, but I don't (yet) know if the current capabilities of `bin/solr zk` can accomplish all of the same things.

**Pull request merged**